src/FSharp.Markdown/MarkdownParser.fs

```diff
@@ -145,107 +146,128 @@ let (|Emphasised|_|) = function
+let rec parseChars acc input (ctx:ParsingContext) = seq {
 
   // Zero or one literals, depending whether there is some accumulated input
   let accLiterals = Lazy.Create(fun () ->
-    if List.isEmpty acc then []
-    else [Literal(String(List.rev acc |> Array.ofList))] )
+    if List.isEmpty acc then ([], ctx)
+    else ([Literal(String(List.rev acc |> Array.ofList), match ctx.CurrentRange with | Some(n) -> Some({ n with EndColumn = n.StartColumn + acc.Length }) | None -> None)], { ctx with CurrentRange = match ctx.CurrentRange with | Some(n) -> Some({ n with StartColumn = n.StartColumn + acc.Length }) | None -> None }) )
 
   match input with
   // Recognizes explicit line-break at the end of line
   | ' '::' '::'\r'::'\n'::rest
   | ' '::' '::('\n' | '\r')::rest ->
-      yield! accLiterals.Value
-      yield HardLineBreak
-      yield! parseChars [] rest
+      let (value, ctx) = accLiterals.Value
+      yield! value
+      yield HardLineBreak(ctx.CurrentRange)
+      yield! parseChars [] rest ctx
 
   // Encode & as an HTML entity
   | '&'::'a'::'m'::'p'::';'::rest
   | '&'::rest ->
-      yield! parseChars (';'::'p'::'m'::'a'::'&'::acc) rest
+      yield! parseChars (';'::'p'::'m'::'a'::'&'::acc) rest ctx
 
   // Ignore escaped characters that might mean something else
   | EscapedChar(c, rest) ->
-      yield! parseChars (c::acc) rest
+      yield! parseChars (c::acc) rest ctx
 
   // Inline code delimited either using double `` or single `
   // (if there are spaces around, then body can contain more backticks)
-  | List.DelimitedWith ['`'; ' '] [' '; '`'] (body, rest)
-  | List.DelimitedNTimes '`' (body, rest) ->
-      yield! accLiterals.Value
-      yield InlineCode(String(Array.ofList body).Trim())
-      yield! parseChars [] rest
+  | List.DelimitedWith ['`'; ' '] [' '; '`'] (body, rest, s, e)
+  | List.DelimitedNTimes '`' (body, rest, s, e) ->
+      let (value, ctx) = accLiterals.Value
+      yield! value
+      yield InlineCode(String(Array.ofList body).Trim(), match ctx.CurrentRange with Some(n) -> Some({ n with StartColumn = n.StartColumn + s; EndColumn = n.EndColumn - e }) | None -> None)
+      yield! parseChars [] rest ctx
 
   // Display Latex inline math mode
   | DelimitedLatexDisplayMath ['$';'$'] (body, rest) ->
-    yield! accLiterals.Value
-    yield LatexDisplayMath(String(Array.ofList body).Trim())
-    yield! parseChars [] rest
+    let (value, ctx) = accLiterals.Value
+    yield! value
+    yield LatexDisplayMath(String(Array.ofList body).Trim(), ctx.CurrentRange)
+    yield! parseChars [] rest ctx
 
   // Inline Latex inline math mode
   | DelimitedLatexInlineMath ['$'] (body, rest) ->
-    yield! accLiterals.Value
-    yield LatexInlineMath(String(Array.ofList body).Trim())
-    yield! parseChars [] rest
+    let (value, ctx) = accLiterals.Value
+    let ctx = { ctx with CurrentRange = match ctx.CurrentRange with | Some(n) -> Some({ n with StartColumn = n.StartColumn + 1 }) | None -> None }
```

Comment:
Maybe these sort of operations could be helper functions, so that you can just say:

    let ctx = ctx.WithRange(incrementStart (0, 1) ctx.Range) // to increment start by `0` lines and `1` columns

Similar operations are done in a few places in the code, so a helper function might be useful (not necessarily exactly as I suggested above - but